packages/debug/src/browser/debug-configuration-manager.ts

```diff
@@ -189,7 +193,26 @@ export class DebugConfigurationManager {
         }
     }
     async addConfiguration(): Promise<void> {
-        const { model } = this;
+        const thismodel = this.model;
+        if (thismodel) {
+            console.log(thismodel);
+        }
+        let model: DebugConfigurationModel | undefined;
+        if (this.models.size > 1) {
```

Comment:
I think this check is not reliable: I was able to construct a case where I'm not asked which folder I would like to configure:

1. Open an unrelated folder
2. Select "Open Workspace" and open a folder looking like this:
```
   Third
     .vscode
       launch.json
```
3. Add a second folder to the workspace looking like this (don't save to a file)
```
   Second
     untitled.txt
```
4. Hit F1->Debug: Add Configuration
5. Observe: I'm not asked which workspace folder to use: the config goes to `Third`

I've been able to trace the problem to the fact that the listener for preference changes is never called when a new workspace root is added: theia/packages/debug/src/browser/debug-configuration-manager.ts#88

I am not really sure what the right fix is, though: should we listen to workspace folder changes or should the preferences system send a change?

Reply:
@akosyakov I'd like your opinion on that one.

Reply:
It sounds fine to listen to workspace root changes, since they are input to `updateModels` function and seems to be easy fix.

But that we don't fire preference events when roots are changed bother me too. We should check what VS Code extensions will expect. Could you file an issue to investigate?